File: Benchmarking/src/gen532.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s4 (
    p0: Vector[Vector[s0]],
    p1: Vector[Vector[s0]]
  )
  case class s7 (
    p0: s0,
    p1: s0
  )
  case class s8 (
    p0: s4,
    p1: Vector[s7]
  )
  def f2(v0: Double): Double = {
    var v3: Double = v0
    var v6: Double = v3
    var v4: Double = v6
    var v7: Double = v0
    var v5: Double = v3
    var v11: Double = v0
    var v8: Double = v4
    val v2: Vector[Double] = Vector(v5, v3, v4, v8, v3, v7)
    val v1: Double = v11 * v8
    val v10: Double = v2(0)
    val v12: Double = v2(5)
    val v9: Double = v2(2)
    var v17: Double = v9
    val v16: Vector[Vector[Double]] = Vector(v2, v2)
    val v19: Vector[Double] = Vector(v12, v6, v17)
    var v43: Double = v1
    var v22: Vector[Double] = v2
    val v35: Vector[Double] = v16(1)
    val v20: Vector[Double] = v16(1)
    v22 = v22.updated(4, v12)
    var v25: Vector[Double] = v19
    val v37: Double = v25(2)
    var v34: Vector[Double] = v22
    var v61: Double = v1
    var v45: Double = v37
    var v40: Vector[Double] = v20
    var v76: Vector[Double] = v2
    var v39: Double = v43
    v34 = v22
    v76 = v35
    val v63: Double = v17 / v39
    val v50: Double = v34(3)
    v40 = v40.updated(2, v12)
    v34 = v34.updated(3, v3)
    val v67: Double = v76(2)
    val v83: Double = v50 / v67
    v22 = v22.updated(4, v63)
    v17 = v45
    v34 = v40
    val v131: Double = v83 / v10
    v5 = v0
    v40 = v40.updated(0, v61)
    v34 = v34.updated(1, v3)
    v131
  }
  @noinline
  def f0(v0: Vector[s8], v1: Double): Double = {
    var v3: Double = v1
    val v7: Double = f2(v1)
    val v5: Vector[Double] = Vector(v1, v7, v1, v1, v3)
    val v24: Double = v5(3)
    val v57: Double = f2(v3)
    v3 = v57
    v24
  }
  def benchmark(): Unit = {
    val v0: Vector[s8] = Vector(s8(s4(Vector(Vector(s0(Vector(Vector(0.0)), Vector(Vector(1.0), Vector(2.0), Vector(3.0)))), Vector(s0(Vector(Vector(4.0)), Vector(Vector(5.0), Vector(6.0), Vector(7.0)))), Vector(s0(Vector(Vector(8.0)), Vector(Vector(9.0), Vector(10.0), Vector(11.0))))), Vector(Vector(s0(Vector(Vector(12.0)), Vector(Vector(13.0), Vector(14.0), Vector(15.0)))), Vector(s0(Vector(Vector(16.0)), Vector(Vector(17.0), Vector(18.0), Vector(19.0)))), Vector(s0(Vector(Vector(20.0)), Vector(Vector(21.0), Vector(22.0), Vector(23.0)))))), Vector(s7(s0(Vector(Vector(24.0)), Vector(Vector(25.0), Vector(26.0), Vector(27.0))), s0(Vector(Vector(28.0)), Vector(Vector(29.0), Vector(30.0), Vector(31.0)))), s7(s0(Vector(Vector(32.0)), Vector(Vector(33.0), Vector(34.0), Vector(35.0))), s0(Vector(Vector(36.0)), Vector(Vector(37.0), Vector(38.0), Vector(39.0)))))), s8(s4(Vector(Vector(s0(Vector(Vector(40.0)), Vector(Vector(41.0), Vector(42.0), Vector(43.0)))), Vector(s0(Vector(Vector(44.0)), Vector(Vector(45.0), Vector(46.0), Vector(47.0)))), Vector(s0(Vector(Vector(48.0)), Vector(Vector(49.0), Vector(50.0), Vector(51.0))))), Vector(Vector(s0(Vector(Vector(52.0)), Vector(Vector(53.0), Vector(54.0), Vector(55.0)))), Vector(s0(Vector(Vector(56.0)), Vector(Vector(57.0), Vector(58.0), Vector(59.0)))), Vector(s0(Vector(Vector(60.0)), Vector(Vector(61.0), Vector(62.0), Vector(63.0)))))), Vector(s7(s0(Vector(Vector(64.0)), Vector(Vector(65.0), Vector(66.0), Vector(67.0))), s0(Vector(Vector(68.0)), Vector(Vector(69.0), Vector(70.0), Vector(71.0)))), s7(s0(Vector(Vector(72.0)), Vector(Vector(73.0), Vector(74.0), Vector(75.0))), s0(Vector(Vector(76.0)), Vector(Vector(77.0), Vector(78.0), Vector(79.0)))))))
    val v1: Double = 80.0
    val start = nanoTime()
    var result: Double = 80.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}

File: Benchmarking/src/gen532.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s4 {
    var p0: [[s0]]
    var p1: [[s0]]
  }
  struct s7 {
    var p0: s0
    var p1: s0
  }
  struct s8 {
    var p0: s4
    var p1: [s7]
  }
  func f2(_ v0: Double) -> Double {
    var v3: Double = v0
    var v6: Double = v3
    var v4: Double = v6
    var v7: Double = v0
    var v5: Double = v3
    var v11: Double = v0
    var v8: Double = v4
    let v2: [Double] = [v5, v3, v4, v8, v3, v7]
    let v1: Double = v11 * v8
    let v10: Double = v2[0]
    let v12: Double = v2[5]
    let v9: Double = v2[2]
    var v17: Double = v9
    let v16: [[Double]] = [v2, v2]
    let v19: [Double] = [v12, v6, v17]
    var v43: Double = v1
    var v22: [Double] = v2
    let v35: [Double] = v16[1]
    let v20: [Double] = v16[1]
    v22[4] = v12
    var v25: [Double] = v19
    let v37: Double = v25[2]
    var v34: [Double] = v22
    var v61: Double = v1
    var v45: Double = v37
    var v40: [Double] = v20
    var v76: [Double] = v2
    var v39: Double = v43
    v34 = v22
    v76 = v35
    let v63: Double = v17 / v39
    let v50: Double = v34[3]
    v40[2] = v12
    v34[3] = v3
    let v67: Double = v76[2]
    let v83: Double = v50 / v67
    v22[4] = v63
    v17 = v45
    v34 = v40
    let v131: Double = v83 / v10
    v5 = v0
    v40[0] = v61
    v34[1] = v3
    return v131
  }
  @inline(never)
  func f0(_ v0: [s8], _ v1: Double) -> Double {
    var v3: Double = v1
    let v7: Double = f2(v1)
    let v5: [Double] = [v1, v7, v1, v1, v3]
    let v24: Double = v5[3]
    let v57: Double = f2(v3)
    v3 = v57
    return v24
  }
  func benchmark() {
    let v0: [s8] = [s8(p0: s4(p0: [[s0(p0: [[0.0]], p1: [[1.0], [2.0], [3.0]])], [s0(p0: [[4.0]], p1: [[5.0], [6.0], [7.0]])], [s0(p0: [[8.0]], p1: [[9.0], [10.0], [11.0]])]], p1: [[s0(p0: [[12.0]], p1: [[13.0], [14.0], [15.0]])], [s0(p0: [[16.0]], p1: [[17.0], [18.0], [19.0]])], [s0(p0: [[20.0]], p1: [[21.0], [22.0], [23.0]])]]), p1: [s7(p0: s0(p0: [[24.0]], p1: [[25.0], [26.0], [27.0]]), p1: s0(p0: [[28.0]], p1: [[29.0], [30.0], [31.0]])), s7(p0: s0(p0: [[32.0]], p1: [[33.0], [34.0], [35.0]]), p1: s0(p0: [[36.0]], p1: [[37.0], [38.0], [39.0]]))]), s8(p0: s4(p0: [[s0(p0: [[40.0]], p1: [[41.0], [42.0], [43.0]])], [s0(p0: [[44.0]], p1: [[45.0], [46.0], [47.0]])], [s0(p0: [[48.0]], p1: [[49.0], [50.0], [51.0]])]], p1: [[s0(p0: [[52.0]], p1: [[53.0], [54.0], [55.0]])], [s0(p0: [[56.0]], p1: [[57.0], [58.0], [59.0]])], [s0(p0: [[60.0]], p1: [[61.0], [62.0], [63.0]])]]), p1: [s7(p0: s0(p0: [[64.0]], p1: [[65.0], [66.0], [67.0]]), p1: s0(p0: [[68.0]], p1: [[69.0], [70.0], [71.0]])), s7(p0: s0(p0: [[72.0]], p1: [[73.0], [74.0], [75.0]]), p1: s0(p0: [[76.0]], p1: [[77.0], [78.0], [79.0]]))])]
    let v1: Double = 80.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 80.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
